frontend/cypress/integration/common/workload_logs.ts

```diff
@@ -0,0 +1,122 @@
+import { Given, Then, When } from "cypress-cucumber-preprocessor/steps";
+
+function goToWorkloadDetailPage(namespace: string, workload: string) {
+    // Go to workloads list
+    cy.get("#Workloads").click();
+
+    // Select "bookinfo" namespace
+    cy.get("#namespace-selector").click();
+    cy.get("#bulk-select-id").uncheck();
+    cy.get(`#namespace-list-item\\[${namespace}\\] input`).check();
+    cy.get("#namespace-selector").click();
+
+    // Open workload detail page
+    cy.get(`[data-test=VirtualItem_Nsbookinfo_${workload}] a`).click();
+}
```

Comment:
> IMO this should be part of the test "Background" in the feature file.

Well... It is not in the Background, but it is part of `Given` clauses. I chose to not use the Background, because explicitly writing the Given clause in each Scenario gives much better clarity to the Scenarios. Also, in the [Gherkin reference, it's mentioned that the background](https://cucumber.io/docs/gherkin/reference/#background) is for "incidental details", but in this case I think this is properly helping to describe the Scenarios.

Anyway, I'm an engineer and I think I somewhat share the idea of DRYing up the code. So, if you tell me that it's better in the Background, I won't put any resistance. So, do you really want that I move it to the Background?

> Something like `Given user is at the "/namespaces/bookinfo/workloads/details-v1?tab=logs" page`

Placing the URL in the sentence makes it handy to reuse the Given clause for other pages. But, again, I'm mainly following the docs and I understand this should be human readable. IMO, a URL becomes technical.

Again, I'm an engineer and I think we won't pass this to non-engineer people. So, if you lean towards usability rather than adhering to the Gherkin language ideas, them I'm quite OK with writing URLs on the sentences. You tell.

> Perhaps testing the link to the logs tab can be a part of the workload details page testing.

I think this is an implementation detail. Simply, it is more hypnotizing the way I have it implemented :-)
But here I give you the reason. I will remove all this code in favor of a simple `cy.visit()`.

Reply:
Personally I see it as Background where you can replace the statement `Given I open Kiali URL` with something like `Given I open the workload details page`. Visiting the root kiali URL is not relevant to this test and just adds more time having to load the page twice.

Regarding the format, I haven't read any of the Gherkin docs and only written a couple tests in this format. I'm fine with whatever format works best for us. If that's a more human readable one, perhaps that can still be parametrized like `Given user is at the "workload details" page on the "logs" tab` or something like that. I see the wording as less important than the implementation. If we can re-use the same implementation across all or most of the tests then that will make writing future tests and maintaining existing ones easier.

Reply:
> Personally I see it as Background where you can replace the statement `Given I open Kiali URL` with something like `Given I open the workload details page`. Visiting the root kiali URL is not relevant to this test and just adds more time having to load the page twice.

Hmm... Your comment made me realize that I probably don't need the `Given I open Kiali URL` in the Background. I think it is enough to have `Given user is at administrator perspective` which AFAIU cannot be replaced. That probably will solve the double-load.

> If that's a more human readable one, perhaps that can still be parametrized like `Given user is at the "workload details" page on the "logs" tab` or something like that.

Well, if we will use this format, I think I prefer to _not_ parametrize and have multiple `Given()` in the code that specifically do the needed `cy.visit()`, rather than a big `switch/if` deciding/forming the URL to visit.

Reply:
> Well, if we will use this format, I think I prefer to not parametrize and have multiple Given() in the code that specifically do the needed cy.visit(), rather than a big switch/if deciding/forming the URL to visit.

Either way is fine with me. The implementations can all share a common function if we want to apply options to all urls like `?refresh=0`. On the downside we'll need to implement more "Given" statements for every feature file. Trading off having more shared step definitions for better readability.

Reply:
OK, I just removed the `Given I open Kiali URL`. Now, the background only has `Given user is at administrator perspective`.

BTW @nrfox, this `goToWorkloadDetailPage` function does no longer exist. Perhaps, can you take another look to see if current state is what you would expect?

I think, at the moment, this is the only blocking comment. So, if you are happy now, I hope we can go ahead and merge.